faviconupdate.js

```diff
@@ -99,67 +98,11 @@ if (SERVICES.hasOwnProperty(awsServiceName)) {
 
 	let awsService = SERVICES[awsServiceName];
 
-	// Ok handle the favicon for a few different situations
-	
-	if (awsServiceName == 'workspaces' || awsServiceName == 'managed-services') {
-
-		// You may be wondering why this code block is exactly the same as the else statement.  That's because for some
-		// unknown reason, `if ('workspaces' == 'cloudsearch' || 'swf')` resolves to true and it is driving me crazy so
-		// I'm just forcing the thing to work here.  Uuuuuugh
-		// Try it though!  Just  delete this block and start the if statement at `if (awsService == 'cloudsearch' || 'swf')`
-		// and see what happens!
-
-		let linkElements = document.getElementsByTagName('link');		
-		for (let i = 0; i < linkElements.length; i++) {
-			if (linkElements[i].getAttribute('rel') == 'icon') {
-				linkElements[i].setAttribute('type', 'image/png');
-				linkElements[i].setAttribute('href', chrome.runtime.getURL(`icons/${awsService.href}`));
-			}
-
-			if (linkElements[i].getAttribute('rel') == 'shortcut icon') {
-				linkElements[i].setAttribute('type', 'image/png');
-				linkElements[i].setAttribute('href', chrome.runtime.getURL(`icons/${awsService.href}`));
-			}
-		}
-
-	} else if (awsService == 'cloudsearch' || 'swf') {
-
-		// In these few cases, we need to actually add the favicon tag, because no one at Amazon did this yet!
-
-		// Build the icon and shortcut icon tags so we can add them inside the head tag
-		let iconNode = document.createElement('link');
-		iconNode.setAttribute('rel', 'icon');
-		iconNode.setAttribute('type', 'image/png');
-		iconNode.setAttribute('href', chrome.runtime.getURL(`icons/${awsService.href}`));
-		let shortcutIconNode = document.createElement('link');
-		shortcutIconNode.setAttribute('rel', 'shortcut icon');
-		shortcutIconNode.setAttribute('type', 'image/png');
-		shortcutIconNode.setAttribute('href', chrome.runtime.getURL(`icons/${awsService.href}`));
-
-		// Add the tags we just made to the head tag
-
-		document.getElementsByTagName('head')[0].appendChild(iconNode);
-		document.getElementsByTagName('head')[0].appendChild(shortcutIconNode);
-
-	} else {
-
-		// We have a favicon on the page, so just update the favicon tags
-
-		// Get all the <link> tags
-		let linkElements = document.getElementsByTagName('link');
-
-		for (let i = 0; i < linkElements.length; i++) {
-
-			// There are 2 tags that control the favicon.  Update them to be the correct favicon
-			if (linkElements[i].getAttribute('rel') == 'icon') {
-				linkElements[i].setAttribute('type', 'image/png');
-				linkElements[i].setAttribute('href', chrome.runtime.getURL(`icons/${awsService.href}`));
-			}
-
-			if (linkElements[i].getAttribute('rel') == 'shortcut icon') {
-				linkElements[i].setAttribute('type', 'image/png');
-				linkElements[i].setAttribute('href', chrome.runtime.getURL(`icons/${awsService.href}`));
-			}
-		}
+	if (awsService) {

```

Comment:
Do we need this if-block? This should always evaluate to true, right?

Reply:
It wouldn't resolve to true when they release a new service that's not in the list, or something like `iam` or `cognito` which are commented out of the list, because amazon already has them.

Reply:
That being said. If/when your PR (#2) goes in this would be redundant.

Reply:
In those cases, it wouldn't reach this inner if-statement at all, right?
Because of this outer if-statement:
https://github.com/JB4GDI/awsfaviconupdater/blob/e382cb943f4f69402d3e00f20e2578239e591cdd/faviconupdate.js#L98

Reply:
Doh you are correct. More code I can delete!